### PR TITLE
Add dev mode nag for recipe IDs that should be synthetic

### LINKED ISSUE #730: Nags for recipe IDs that should be synthetic
As per the EmiRecipe javadoc
https://github.com/emilyploszaj/emi/blob/6127e8f6a9b736140c384f58d20faadd61add2db/xplat/src/main/java/dev/emi/emi/api/recipe/EmiRecipe.java#L23

Check that every registered EmiRecipe with a non-synthetic ID actually exists by looking it up in the recipe manager. Log those that don't.

### PR DESCRIPTION
**Pull request by Abbie5**
Closes #730

Also changes some brewing recipe IDs to be synthetic.

### CODE REVIEW

File: xplat/src/main/java/dev/emi/emi/screen/WidgetGroup.java [hunk 1]
```diff
@@ -79,6 +79,10 @@ public void decorateDevMode() {
 				tooltip.addAll(EmiTooltip.splitTranslate("emi.dev.synthetic_id"));
 			}
 			errors.add(new RecipeError(RecipeError.Severity.ERROR, tooltip));
+		} else if (EmiDev.missingRecipeIds.contains(id)) {
+			List<TooltipComponent> tooltip = Lists.newArrayList();
+			tooltip.addAll(EmiTooltip.splitTranslate("emi.dev.synthetic_id"));
```
Comment (emilyploszaj) [inline xplat/src/main/java/dev/emi/emi/screen/WidgetGroup.java:85]: Please add a more detailed tooltip for what's going on and why a dev should use a synthetic ID.

File: xplat/src/main/java/dev/emi/emi/runtime/dev/EmiDev.java [hunk 2]
```diff
@@ -6,4 +6,5 @@
 
 public class EmiDev {
 	public static Set<Identifier> duplicateRecipeIds = Set.of();
+	public static Set<Identifier> missingRecipeIds = Set.of();
```
Comment (emilyploszaj) [inline xplat/src/main/java/dev/emi/emi/runtime/dev/EmiDev.java]: All instances of `missingRecipeIds` and similar should probably be renamed, they're not missing IDs (i.e. null). Can't think of a great name, but `incorrectRecipeIds` would be better, if you have something you think suits it better go for it.

File: xplat/src/main/java/dev/emi/emi/registry/EmiRecipes.java [hunk 3]
```diff
@@ -178,13 +179,20 @@ public Manager(List<EmiRecipeCategory> categories, Map<EmiRecipeCategory, List<E
...
+					}
 				}
 			}
 	
 			if (EmiConfig.devMode) {
 				for (Identifier id : duplicateIds.keySet()) {
 					EmiReloadLog.warn(duplicateIds.getInt(id) + " recipes loaded with the same id: " + id);
 				}
+				for (Identifier id : missingIds) {
+					EmiReloadLog.warn("Recipe " + id + " not present in recipe manager. Consider prefixing its path with '/' to signify that it is synthetic.");
```
Comment (emilyploszaj) [inline xplat/src/main/java/dev/emi/emi/registry/EmiRecipes.java]: Nit: perhaps change to "Consider prefixing its path with '/' if it is synthetic"

**Review by emilyploszaj — CHANGES_REQUESTED** [review]
Thanks for the contribution and for finding all of the non-synthetic IDs remaining in EMI. Few crits then I'll merge.

**Comment by Abbie5** [comment]
Sorry for taking so long, I've made the requested changes now.

**Review by emilyploszaj — APPROVED** [review]

**Pull request merged**